fetch-array-buffer/index.html

```diff
@@ -23,59 +20,55 @@ <h1>Fetch arrayBuffer example</h1>
     <pre></pre>
   </body>
   <script>
-    // define variables
-    var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
-    var source;
-
-    var pre = document.querySelector('pre');
-    var myScript = document.querySelector('script');
-    var play = document.querySelector('.play');
-    var stop = document.querySelector('.stop');
-    var errorDisplay = document.querySelector('.error');
+    const pre = document.querySelector("pre");
+    const myScript = document.querySelector("script");
+    const play = document.querySelector(".play");
+    const stop = document.querySelector(".stop");
+    const errorDisplay = document.querySelector(".error");
 
     // use fetch to load an audio track, and
     // decodeAudioData to decode it and stick it in a buffer.
     // Then we put the buffer into the source
     function getData() {
-      source = audioCtx.createBufferSource();
+      const source = new AudioBufferSourceNode();
 
-      return fetch('viper.ogg')
-      .then(function(response) {
-        if (!response.ok) {
-          throw new Error("HTTP error, status = " + response.status);
-        }
-        return response.arrayBuffer();
-      })
-      .then(function(buffer) {
-        audioCtx.decodeAudioData(buffer, function(decodedData) {
-          source.buffer = decodedData;
-          source.connect(audioCtx.destination);
+      return fetch("viper.ogg")
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error(`HTTP error, status = ${response.status}`);
+          }
+          return response.arrayBuffer();
+        })
+        .then((buffer) => {
+          audioCtx.decodeAudioData(buffer).then((decodedData) => {
+            source.buffer = decodedData;
+            source.connect(audioCtx.destination);
+            return source;
+          });
         });
```

Comment:
`audioCtx` is undefined now—also we need to return this promise because we have a nested promise here

Reply:
Good catch.